sourcegraph/sourcegraph
# Wildcard [Popover]: Adjust `<Popover />` API with possible future needs

Closes https://github.com/sourcegraph/sourcegraph/issues/36212

## Background 

Based on this PR https://github.com/sourcegraph/sourcegraph/pull/35870 about creation `<Tooltip/>` component we are going to have a part of radix-ui - [tooltip component](https://www.radix-ui.com/docs/primitives/components/tooltip). This radix Tooltip component has a compound like API. 

```tsx
 <Tooltip.Provider>
    <Tooltip.Root>
      <Tooltip.Trigger />
      <Tooltip.Content>
        <Tooltip.Arrow />
      </Tooltip.Content>
    </Tooltip.Root>
  </Tooltip.Provider>
```

Since It's possible that in the (near) future we would try to replace the radix tooltip with our own implementation of the floating UI - Popover component, this PR updates the` <Popover />` component API in order to have a similar API with radix-ui `<Tooltip />. To be more specific in this PR we rework the `<PopoverTail />` component API and refactor the popover implementation by splitting it into separate compound components. 

The next step here would be trying to come up with a minimal example/demo of our own `<Tooltip />` component with all vital a11y requirements. 

## Test plan
- Make sure that any consumers of the Popover components don't have any regression around positions and tail styling. 
- Make sure that the browser extension popover element looks good on the repository page



## App preview:

- [Web](https://sg-web-vk-update-popover-api.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-dpegqgsiki.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
